store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -6219,6 +6221,60 @@ private void checkPasswordStrength(String password, Account acct, Cos cos, ZMuta
         }
     }
 
+  /**
+   * Validate user password and fails if passed password string contains any of their personal data
+   *
+   * @param password Password string against which checks to be performed
+   * @param acct {@link Account} from where personal data like:
+   *     last name({@value Provisioning#A_sn});
+   *     common name({@value Provisioning#A_cn});
+   *     initials of some or all names, but not the surname({@value Provisioning#A_initials})
+   *     and email address({@value Provisioning#A_mail}) will be compared
+   * @throws AccountServiceException
+   */
+  private void validatePasswordEntropyForPersonalData(String password, Account acct)
+      throws AccountServiceException {
+    // get possible personal data for account
+    Optional<String> sName =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_sn)).orElse(""));
+    Optional<String> cName =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_cn)).orElse(""));
+    Optional<String> initials =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_initials)).orElse(""));
+    Optional<String> email =
+      Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_mail)).orElse(""));
```

Comment:
Check out `provisioningTest`

Reply:
What ?

Reply:
I mean I don't understand why `Optional.of(Optional.ofNullable(...).orElse(""))` when you can write this way:
```suggestion
    Optional<String> sName = Optional.ofNullable(acct.getAttr(Provisioning.A_sn));
    Optional<String> cName = Optional.ofNullable(acct.getAttr(Provisioning.A_cn));
    Optional<String> initials = Optional.ofNullable(acct.getAttr(Provisioning.A_initials));
    Optional<String> email = Optional.ofNullable(acct.getAttr(Provisioning.A_mail));
```

The behavior should be the same

Alternatively you can do the following:
```suggestion
    String sName = Optional.ofNullable(acct.getAttr(Provisioning.A_sn)).orElse("");
    String cName = Optional.ofNullable(acct.getAttr(Provisioning.A_cn)).orElse("");
    String initials = Optional.ofNullable(acct.getAttr(Provisioning.A_initials)).orElse("");
    String email = Optional.ofNullable(acct.getAttr(Provisioning.A_mail)).orElse("");
```

If you want the "fallback" value to be `""`

Reply:
Offcourse.